src/python/pants/engine/target.py

```diff
@@ -2450,9 +2450,8 @@ class Dependencies(StringSequenceField, AsyncFieldMixin):
         `{bin_name()} dependencies` or `{bin_name()} peek` on this target to get the final
         result.
 
-        See {doc_url('targets#target-addresses')} and {doc_url('targets#target-generation')} for
-        more about how addresses are formed, including for generated targets. You can also run
-        `{bin_name()} list ::` to find all addresses in your project, or
+        See {doc_url('targets')} for more about how addresses are formed, including for generated
+        targets. You can also run `{bin_name()} list ::` to find all addresses in your project, or
         `{bin_name()} list dir:` to find all addresses defined in that directory.
```

Comment:
Unrelated to your change, but:

```suggestion
        `{bin_name()} list dir` to find all addresses defined in that directory.
```

Otherwise we will expand target generators w/ all their generated targets, which IMO is not desirable. We generally don't mention `:` much in docs anymore, either, so it's confusing to use here.